File: scratch/inspect_2006_dep.py
import zipfile
import json
import os

zip_path = 'resultados_geo/Legislativas 2006/deputados_federal_2006_RJ.zip'
with zipfile.ZipFile(zip_path, 'r') as z:
    for filename in z.namelist():
        if filename.endswith('.json'):
            print('File found inside zip:', filename)
            with z.open(filename) as f:
                data = json.loads(f.read().decode('utf-8'))
                
                print('METADATA keys:', list(data.get('METADATA', {}).keys()))
                cand_names = data.get('METADATA', {}).get('cand_names', {})
                print('Number of candidates in cand_names:', len(cand_names))
                cand_keys = list(cand_names.keys())
                print('Sample candidates (first 10):', cand_keys[:10])
                for k in cand_keys[:10]:
                    print(f'cand_names[{k}]:', cand_names[k])
                
                coalition = data.get('METADATA', {}).get('coalition_adjustments', {})
                print('Sample coalition_adjustments (first 5):', list(coalition.items())[:5])
                
                results = data.get('RESULTS', {})
                print('Number of results:', len(results))
                result_keys = list(results.keys())
                print('Sample RESULTS keys (first 3):', result_keys[:3])
                for k in result_keys[:3]:
                    print(f'RESULTS[{k}]:', results[k])

File: scratch/inspect_2006_dep.js
const fs = require('fs');
const path = require('path');
const admZip = require('adm-zip');

const zipPath = 'resultados_geo/Legislativas 2006/deputados_federal_2006_RJ.zip';
const zip = new admZip(zipPath);
const zipEntries = zip.getEntries();

zipEntries.forEach((entry) => {
  if (entry.entryName.endsWith('.json')) {
    console.log('File found inside zip:', entry.entryName);
    const content = entry.getData().toString('utf8');
    const json = JSON.parse(content);
    
    console.log('METADATA structure keys:', Object.keys(json.METADATA || {}));
    if (json.METADATA && json.METADATA.cand_names) {
      const candKeys = Object.keys(json.METADATA.cand_names);
      console.log('Sample cand_names keys (first 10):', candKeys.slice(0, 10));
      candKeys.slice(0, 10).forEach(k => {
        console.log(`cand_names[${k}]:`, json.METADATA.cand_names[k]);
      });
    }

    if (json.METADATA && json.METADATA.coalition_adjustments) {
      console.log('coalition_adjustments (first 5):', Object.entries(json.METADATA.coalition_adjustments).slice(0, 5));
    }
    
    if (json.RESULTS) {
      const resultKeys = Object.keys(json.RESULTS);
      console.log('Sample RESULTS keys (first 5):', resultKeys.slice(0, 5));
      resultKeys.slice(0, 3).forEach(k => {
        console.log(`RESULTS[${k}]:`, json.RESULTS[k]);
      });
    }
  }
});
